rasa_core/policies/embedding_policy.py

```diff
@@ -0,0 +1,2043 @@
+                 output_attention=False,
+                 initial_cell_state=None,
+                 name=None):
+        super(TimeAttentionWrapper, self).__init__(
+                cell,
+                attention_mechanism,
+                attention_layer_size,
+                alignment_history,
+                cell_input_fn,
+                output_attention,
+                initial_cell_state,
+                name
+        )
+        self._attn_input_fn = attn_input_fn
+
+        self._dialogue_len = dialogue_len
+        if not isinstance(attn_shift_range, list):
+            attn_shift_range = [attn_shift_range]
+        self._timed_ntms = [TimedNTM(attn_shift_range[0],
+                                     sparse_attention,
+                                     dialogue_len,
+                                     name='0')]
+        if self._is_multi:
+            for i in range(1, len(attention_mechanism)):
+                if len(attn_shift_range) < i + 1:
+                    attn_shift_range.append(attn_shift_range[-1])
+                self._timed_ntms.append(TimedNTM(attn_shift_range[i],
+                                                 sparse_attention,
+                                                 dialogue_len,
+                                                 name=str(i)))
+        self._sim_fn = similarity_fn
+        self._emb_for_action_listen = emb_for_action_listen
+        if not isinstance(ignore_mask, list):
+            self._ignore_mask = [ignore_mask]
+        else:
+            self._ignore_mask = ignore_mask
+
+        self._copy_attn_to_out = copy_attn_to_out
+
+    @property
+    def output_size(self):
+        if self._output_attention:
+            return (self._attention_layer_size
+                    + 2
+                    * self._cell.output_size
+                    + 2
+                    + 4)
+        else:
+            return self._cell.output_size
+
+    @property
+    def state_size(self):
+        """The `state_size` property of `AttentionWrapper`.
+        Returns:
+          An `AttentionWrapperState` tuple containing shapes used by this
+          object.
+        """
+
+        if isinstance(self._cell.state_size, tf.contrib.rnn.LSTMStateTuple):
+            hidden_state_size = self._cell.state_size.c
+        else:
+            hidden_state_size = self._cell.state_size
+
+        return TimeAttentionWrapperState(
+                cell_state=self._cell.state_size,
+                time=tf.TensorShape([]),
+                attention=self._attention_layer_size,
+                alignments=self._item_or_tuple(
+                        a.alignments_size for a in self._attention_mechanisms),
+                attention_state=self._item_or_tuple(
+                        a.state_size for a in self._attention_mechanisms),
+                alignment_history=self._item_or_tuple(
+                        a.alignments_size if self._alignment_history else ()
+      
```

Comment:
I did that, because it was faster for me to copy, to actually check if it was working. I'm not sure how to work with `collections` and how to add one more element into the tuple. Also RE subclassing `AttentionWrapperState`